Define and activate KVM 12.2.0 platform release (#384)
giantswarm/releases: kvm/v12.2.0/README.md

## :zap:  Giant Swarm Release 12.2.0 for KVM :zap:

**If you are upgrading from 12.1.0, upgrading to this release will not roll your nodes.**

> As of this release **NGINX Ingress Controller App** is **optional and not pre-installed** component on KVM. This allows NGINX App installations to be managed independently from the base platform lifecycle. It is both benefit but also new responsibility to keep NGINX App installations up-to-date separately from rest of the cluster. Making NGINX optional enables use of other ingress controller alternatives without wasting resources where NGINX is not the preferred option. Upgrading existing tenant clusters with pre-installed NGINX will leave NGINX unchanged. Existing NGINX App custom resources will still have `giantswarm.io/managed-by: cluster-operator` label, but it will be ignored. The label will be cleaned up at a later point after all tenant clusters have been upgraded and KVM platform releases older than v12.2.0 archived.

Reviewer: ```suggestion
As of this release, **NGINX Ingress Controller App** is now an **optional (and not pre-installed)** component on KVM.
```

> As of this release **NGINX Ingress Controller App** is **optional and not pre-installed** component on KVM. This allows NGINX App installations to be managed independently from the base platform lifecycle. It is both benefit but also new responsibility to keep NGINX App installations up-to-date separately from rest of the cluster. Making NGINX optional enables use of other ingress controller alternatives without wasting resources where NGINX is not the preferred option. Upgrading existing tenant clusters with pre-installed NGINX will leave NGINX unchanged. Existing NGINX App custom resources will still have `giantswarm.io/managed-by: cluster-operator` label, but it will be ignored. The label will be cleaned up at a later point after all tenant clusters have been upgraded and KVM platform releases older than v12.2.0 archived.

Reviewer: ```suggestion
As of this release, **NGINX Ingress Controller App** is now an **optional (and not pre-installed)** component on KVM.

This enables use of alternative ingress controllers without wasting resources where NGINX is not the preferred option.

It also allows NGINX App installations to be managed independently from the base platform lifecycle, making it easier and less disruptive to update the app without affecting your nodes.

Upgrading existing tenant clusters with pre-installed NGINX will leave NGINX unchanged. Existing NGINX App custom resources will still have `giantswarm.io/managed-by: cluster-operator` label, but it will be ignored. The label will be cleaned up at a later point after all tenant clusters have been upgraded and KVM platform releases older than v12.2.0 archived.
```

Reviewer: @sslavic super nice release notes draft thank you! Here are the changes I'm suggesting:

1. Broke it down into paragraphs to make it easier to read
2. "It also allows NGINX App installations to be managed independently from the base platform lifecycle, making it easier and less disruptive to update the app without affecting your nodes." - tried to frame it into why customer benefits
3. "It is both benefit but also new responsibility to keep NGINX App installations up-to-date separately from rest of the cluster." - this seemed out of place for release notes... wdyt?

Author: > making it easier and less disruptive to update the app without affecting your nodes.

that was the case before as well, when platform release had apps only, so it's not unique to nginx being distributed as optional app.

I copied the release notes from Azure 12.0.0. That release was more packed, so this part for NGINX was compressed into single paragraph. Here in KVM 12.2.0 it makes sense to spread it a bit for readability.

Reviewer: @sslavic what about this for the first 3 paragraphs?

---

As of this release, **NGINX Ingress Controller App** is now an **optional (and not pre-installed)** component on KVM.

This enables use of alternative ingress controllers without wasting resources where NGINX is not the preferred option.

Now NGINX App installations can be managed and updated independent of the cluster, which is both a benefit and a new responsibility 😅

Author: LGTM

> As of this release **NGINX Ingress Controller App** is **optional and not pre-installed** component on KVM. This allows NGINX App installations to be managed independently from the base platform lifecycle. It is both benefit but also new responsibility to keep NGINX App installations up-to-date separately from rest of the cluster. Making NGINX optional enables use of other ingress controller alternatives without wasting resources where NGINX is not the preferred option. Upgrading existing tenant clusters with pre-installed NGINX will leave NGINX unchanged. Existing NGINX App custom resources will still have `giantswarm.io/managed-by: cluster-operator` label, but it will be ignored. The label will be cleaned up at a later point after all tenant clusters have been upgraded and KVM platform releases older than v12.2.0 archived.

Reviewer: ```suggestion
As of this release, **NGINX Ingress Controller App** is now an **optional (and not pre-installed)** component on KVM.

This enables use of alternative ingress controllers without wasting resources where NGINX is not the preferred option.

Now NGINX App installations can be managed and updated independent of the cluster, which is both a benefit and a new responsibility 😅

Upgrading tenant clusters with pre-installed NGINX will leave NGINX unchanged. Existing NGINX App custom resources will still have `giantswarm.io/managed-by: cluster-operator` label, but it will be ignored. The label will be cleaned up later after all tenant clusters have been upgraded and KVM platform releases older than v12.2.0 archived.
```
